Tests/applicationInsights.tests.ts

```diff
@@ -0,0 +1,64 @@
+///<reference path='..\Declarations\node\node.d.ts' />
+///<reference path='..\Declarations\mocha\mocha.d.ts' />
+///<reference path='..\Declarations\sinon\sinon.d.ts' />
+
+import assert = require("assert");
+import sinon = require("sinon");
+
+describe('AppInsights', () => {
+
+    var warnSpy;
+    before(() => warnSpy = sinon.spy(console, "warn"));
+    after(() => warnSpy.restore());
+
+    describe('#setup()', () => {
+        var AppInsights;
+
+        beforeEach(() => {
+            warnSpy.reset();
+            AppInsights = require("../ApplicationInsights");
+        });
+        
+        afterEach(() => AppInsights.instance = undefined);
+        after(() => warnSpy.restore());
+
+        it('should not warn if setup is called once', () => {
+            AppInsights.setup("key");
+            assert.ok(warnSpy.notCalled, "warning was not raised");
+        });
+
+        it('should warn if setup is called twice', () => {
+            AppInsights.setup("key");
```

Comment:
And super random consistency question/pt: not sure why `"key"` is using double quotes mixed with single quotes.